### PR TITLE
can:mcux_flexcan driver errata

### LINKED ISSUE #31555: tests:drivers_can_api: mimxrt1060 can api test meet assert failure
**Describe the bug**
test_filter_attach ASSERTION FAIL [!FLEXCAN_IsMbOccupied(base, mbIdx)] @ WEST_TOPDIR/modules/hal/nxp/mcux/drivers/imx/fsl_flexcan.c:1734 E: r0/a1: 0x00000004 r1/a2: 0x000006c6 r2/a3: 0x00800000 E: r3/a4: 0x60004121 r12/ip: 0x00000000 r14/lr: 0x600064a3 E: xpsr: 0x41000000 E: Faulting instruction address (r15/pc): 0x6000a876 E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0 E: Current thread: 0x80000218 (unknown) E: Halting system </error>

**To Reproduce**
1. mkdir build; cd build
2. cmake -DBOARD=mimxrt1060_evk ..
3. make flash --runner=pyocd

**Expected behavior**
test sucessfully

**Impact**
can operation

**Logs and console output**
```
test_filter_attach ASSERTION FAIL [!FLEXCAN_IsMbOccupied(base, mbIdx)] @ WEST_TOPDIR/modules/hal/nxp/mcux/drivers/imx/fsl_flexcan.c:1734 E: r0/a1: 0x00000004 r1/a2: 0x000006c6 r2/a3: 0x00800000 E: r3/a4: 0x60004121 r12/ip: 0x00000000 r14/lr: 0x600064a3 E: xpsr: 0x41000000 E: Faulting instruction address (r15/pc): 0x6000a876 E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0 E: Current thread: 0x80000218 (unknown) E: Halting system </error>
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, )
 - Toolchain (Zephyr SDK, ...)
 - Commit SHA or Version used:
 zephyr-v2.4.0-3224-gc77ae15c9967

### LINKED ISSUE #31741: tests:subsys_canbus_isotp: mimxrt1060 meet recv timeout
**Describe the bug**
after apply the PR https://github.com/zephyrproject-rtos/zephyr/pull/31730 ,  we meet data derived issue in isotp 

**To Reproduce**
set up zephyr build and code from https://docs.zephyrproject.org/latest/getting_started/index.html 
1. west init .
2. west update
3. cd zephyr
4. source ./zephyr-env.sh
5. cd tests/subsys/canbus/isotp/conformance
6. west build -b mimxrt1060_evk  -- -G'Unix Makefiles'
7. connect mimxrt1060_evk board with usb serial to host pc
8. west flash --runner=pyocd

**Expected behavior**
Test PASS

**Impact**
ISOTP can stack supporting

**Logs and console output**
```
Running test suite isotp_conformance
===================================================================
START - test_send_sf
 PASS - test_send_sf
===================================================================
START - test_receive_sf
I: SF DL does not match. Ignore
I: SF DL does not match. Ignore
 PASS - test_receive_sf
===================================================================
START - test_send_sf_ext
 PASS - test_send_sf_ext
===================================================================
START - test_receive_sf_ext
I: SF DL does not match. Ignore
W: Unhandled error/status (status 0x00000001, result = 0x00000001
 PASS - test_receive_sf_ext
===================================================================
START - test_send_data
 PASS - test_send_data
===================================================================
START - test_send_data_blocks
 PASS - test_send_data_blocks
===================================================================
START - test_receive_data
 PASS - test_receive_data
===================================================================
START - test_receive_data_blocks
 PASS - test_receive_data_blocks
===================================================================
START - test_send_timeouts
E: Reception of next FC has timed out
E: Reception of next FC has timed out
E: Reception of next FC has timed out
 PASS - test_send_timeouts
===================================================================
START - test_receive_timeouts
E: Timeout while waiting for CF
 PASS - test_receive_timeouts
===================================================================
START - test_stmin
 PASS - test_stmin
===================================================================
START - test_receiver_fc_errors
E: Sequence number missmatch
W: Unhandled error/status (status 0x00000001, result = 0x00000002
 PASS - test_receiver_fc_errors
===================================================================
START - test_sender_fc_errors
W: Unhandled error/status (status 0x00000001, result = 0x00000002

    Assertion failed at WEST_TOPDIR/zephyr/tests/subsys/canbus/isotp/conformance/src/main.c:833: test_sender_fc_errors: (ret not equal to 0)
Send complete callback not called
 FAIL - test_sender_fc_errors
===================================================================
Test suite isotp_conformance failed.

```
**Environment (please complete the following information):**
 - OS: (e.g. Linux)
 - Toolchain (e.g Zephyr SDK, ...)

### PR DESCRIPTION
**Pull request by hakehuang**
in NXP flexcan driver we need add ERRATA for
ERRATA 5461 and 5829, reproted in #31555

Signed-off-by: Hake Huang <hake.huang@oss.nxp.com>

### CODE REVIEW

**Comment by hakehuang** [comment]
for tests/drivers/can/api 
```
*** Booting Zephyr OS build v2.5.0-rc1-107-gaf67564573f5  ***
Running test suite can_driver
===================================================================
START - test_set_loopback
 PASS - test_set_loopback
===================================================================
START - test_send_and_forget
 PASS - test_send_and_forget
===================================================================
START - test_filter_attach
 PASS - test_filter_attach
===================================================================
START - test_receive_timeout
 PASS - test_receive_timeout
===================================================================
START - test_send_callback
 PASS - test_send_callback
===================================================================
START - test_send_receive_std
 PASS - test_send_receive_std
===================================================================
START - test_send_receive_ext
 PASS - test_send_receive_ext
===================================================================
START - test_send_receive_std_masked
 PASS - test_send_receive_std_masked
===================================================================
START - test_send_receive_ext_masked
 PASS - test_send_receive_ext_masked
===================================================================
START - test_send_receive_buffer
 PASS - test_send_receive_buffer
===================================================================
START - test_send_receive_wrong_id
 PASS - test_send_receive_wrong_id
===================================================================
START - test_send_invalid_dlc
E: DLC of 9 exceeds maximum (8)
 PASS - test_send_invalid_dlc
===================================================================
Test suite can_driver succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL

```

**Comment by hakehuang** [comment]
*** Booting Zephyr OS build v2.5.0-rc1-108-ga3c2e2f92091  ***
Running test suite isotp_conformance
===================================================================
START - test_send_sf
 PASS - test_send_sf
===================================================================
START - test_receive_sf
I: SF DL does not match. Ignore
I: SF DL does not match. Ignore
 PASS - test_receive_sf
===================================================================
START - test_send_sf_ext
 PASS - test_send_sf_ext
===================================================================
START - test_receive_sf_ext
I: SF DL does not match. Ignore
I: SF DL does not match. Ignore
 PASS - test_receive_sf_ext
===================================================================
START - test_send_data
 PASS - test_send_data
===================================================================
START - test_send_data_blocks
 PASS - test_send_data_blocks
===================================================================
START - test_receive_data
 PASS - test_receive_data
===================================================================
START - test_receive_data_blocks
 PASS - test_receive_data_blocks
===================================================================
START - test_send_timeouts
E: Reception of next FC has timed out
E: Reception of next FC has timed out
E: Reception of next FC has timed out
 PASS - test_send_timeouts
===================================================================
START - test_receive_timeouts
E: Timeout while waiting for CF
 PASS - test_receive_timeouts
===================================================================
START - test_stmin
 PASS - test_stmin
===================================================================
START - test_receiver_fc_errors
E: Sequence number missmatch
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
 PASS - test_receiver_fc_errors
===================================================================
START - test_sender_fc_errors
E: Pkt length is 5118 but buffer has only 768 bytes
E: Got overflow FC frame
E: Got overflow FC frame
I: Got to many wait frames
 PASS - test_sender_fc_errors
===================================================================
Test suite isotp_conformance succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL
W: Bitrate error: 40
*** Booting Zephyr OS build v2.5.0-rc1-108-ga3c2e2f92091  ***
Running test suite isotp_conformance
===================================================================
START - test_send_sf
 PASS - test_send_sf
===================================================================
START - test_receive_sf
I: SF DL does not match. Ignore
I: SF DL does not match. Ignore
 PASS - test_receive_sf
===================================================================
START - test_send_sf_ext
 PASS - test_send_sf_ext
===================================================================
START - test_receive_sf_ext
I: SF DL does not match. Ignore
I: SF DL does not match. Ignore
 PASS - test_receive_sf_ext
===================================================================
START - test_send_data
 PASS - test_send_data
===================================================================
START - test_send_data_blocks
 PASS - test_send_data_blocks
===================================================================
START - test_receive_data
 PASS - test_receive_data
===================================================================
START - test_receive_data_blocks
 PASS - test_receive_data_blocks
===================================================================
START - test_send_timeouts
E: Reception of next FC has timed out
E: Reception of next FC has timed out
E: Reception of next FC has timed out
 PASS - test_send_timeouts
===================================================================
START - test_receive_timeouts
E: Timeout while waiting for CF
 PASS - test_receive_timeouts
===================================================================
START - test_stmin
 PASS - test_stmin
===================================================================
START - test_receiver_fc_errors
E: Sequence number missmatch
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
I: Got unexpected frame. Ignore
 PASS - test_receiver_fc_errors
===================================================================
START - test_sender_fc_errors
E: Pkt length is 5118 but buffer has only 768 bytes
E: Got overflow FC frame
E: Got overflow FC frame
I: Got to many wait frames
 PASS - test_sender_fc_errors
===================================================================
Test suite isotp_conformance succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL

**Comment by hakehuang** [comment]
isotp_implement log

```
*** Booting Zephyr OS build v2.5.0-rc1-109-gaee9e6364e3f  ***
Running test suite isotp
===================================================================
START - test_bind_unbind
 PASS - test_bind_unbind
===================================================================
START - test_send_receive_net_sf
 PASS - test_send_receive_net_sf
===================================================================
START - test_send_receive_net_blocks
 PASS - test_send_receive_net_blocks
===================================================================
START - test_send_receive_net_single_blocks
 PASS - test_send_receive_net_single_blocks
===================================================================
START - test_send_receive_sf
 PASS - test_send_receive_sf
===================================================================
START - test_send_receive_blocks
 PASS - test_send_receive_blocks
===================================================================
START - test_send_receive_single_block
 PASS - test_send_receive_single_block
===================================================================
START - test_buffer_allocation
 PASS - test_buffer_allocation
===================================================================
START - test_buffer_allocation_wait
 PASS - test_buffer_allocation_wait
===================================================================
Test suite isotp succeeded
===================================================================
PROJECT EXECUTION SUCCESSFUL

**Review by henrikbrixandersen — APPROVED** [review]
LGTM

**Review by MaureenHelm — COMMENTED** [review]
Thanks @hakehuang !

The fix looks good, but please address the compliance failure (leading space).

@nashif the doc failure looks unrelated

**Comment by hakehuang** [comment]
> Thanks @hakehuang !
> 
> The fix looks good, but please address the compliance failure (leading space).

thanks maureen. updated

File: tests/subsys/canbus/isotp/conformance/src/main.c [hunk 1]
```diff
@@ -801,6 +801,7 @@ void test_receiver_fc_errors(void)
 	can_detach(can_dev, filter_id);
 	k_msgq_cleanup(&frame_msgq);
 	isotp_unbind(&recv_ctx);
+	k_sleep(K_SECONDS(1));
```
Comment (alexanderwachter) [inline tests/subsys/canbus/isotp/conformance/src/main.c:804]: Please remove this change. We have another PR that will fix it

**Review by alexanderwachter — CHANGES_REQUESTED** [review]

File: tests/subsys/canbus/isotp/conformance/src/main.c (hunk 1 above)
Comment (MaureenHelm) [reply tests/subsys/canbus/isotp/conformance/src/main.c:804]: @alexanderwachter Can you give a link to that PR?

**Review by MaureenHelm — APPROVED** [review]

**Comment by alexanderwachter** [comment]
#32078

File: tests/subsys/canbus/isotp/conformance/src/main.c (hunk 1 above)
Comment (hakehuang) [reply tests/subsys/canbus/isotp/conformance/src/main.c]: > #31078

@alexanderendian and @MaureenHelm  I remove this fixes, Thanks

**Review by alexanderwachter — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**